p2p/src/lib.rs

```diff
@@ -94,3 +94,57 @@ impl From<io::Error> for Error {
 
 /// Result shorthand.
 pub type Result<T, E = Error> = core::result::Result<T, E>;
+
+pub(crate) mod unbounded_with_len {
```

Comment:
Can you add documentation for this module, functions and structs inside it? Also I thought `clippy` will warn about it

Reply:
It's a private module. It doesn't need docs, but extra docs don't hurt.